src/V3Case.cpp

```diff
@@ -274,8 +310,10 @@ class CaseVisitor final : public VNVisitor {
         // Convert valueItem from AstCaseItem* to the expression
         // Not done earlier, as we may now have a nullptr because it's just a ";" NOP branch
         for (uint32_t i = 0; i < numCases; ++i) {
-            if (AstCaseItem* const itemp = VN_AS(m_valueItem[i], CaseItem)) {
-                m_valueItem[i] = itemp->stmtsp();
+            if (AstNode* const condp = m_valueItem[i]) {
+                AstCaseItem* caseItemp = VN_CAST(condp, CaseItem);
+                if (!caseItemp) caseItemp = VN_AS(condp->abovepIter(), CaseItem);
```

Comment:
I'd prefer not to add a new AstNode base iterator just for this.  Alternatives:
* Build a map earlier to map the condp back to the CaseItem
* Backtrack searching for the CaseItem here.

Reply:
Backtrack searching requires `m_headtailp` which is a private member of `AstNode`, and I don't want to add another 512kB `std::array<AstNode*, 1 << CASE_OVERLAP_WIDTH> m_condItem`, so I use a map now.